website/routes.py

```diff
@@ -765,6 +765,12 @@ def make_url_map(app):
         ], 'get', project_views.node.node_registrations,
             OsfWebRenderer('project/registrations.mako')),
 
+        Rule([
+            '/project/<pid>/retract_registration/',
```

Comment:
@brianjgeiger But `retraction` isn't as much as resource as it is an action on a `project`. `retract` could be considered a shortcut for doing `PUT /project/<pid>/ {is_retracted: true,...}`.